File: services/auth-service/src/main/java/com/zerofake/auth/service/AuthService.java
package com.zerofake.auth.service;

import com.zerofake.auth.dto.common.ApiResponse;
import com.zerofake.auth.dto.request.LoginRequest;
import com.zerofake.auth.dto.request.RefreshTokenRequest;
import com.zerofake.auth.dto.request.RegisterRequest;
import com.zerofake.auth.dto.response.AuthResponse;
import com.zerofake.auth.dto.response.RegisterResponse;
import com.zerofake.auth.dto.response.TokenResponse;
import com.zerofake.auth.dto.response.UserResponse;

public interface AuthService {

    RegisterResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    TokenResponse refreshToken(RefreshTokenRequest request);

    ApiResponse<Void> logout();

    UserResponse getCurrentUser();
}

File: zerofake-frontend/src/services/authService.ts
import { authApi } from "@/api/authApi";

import type {
  AuthResponse,
  LoginRequest,
  RefreshTokenRequest,
  TokenResponse,
  UserResponse,
  RegisterRequest,
  RegisterResponse,
} from "@/types/auth";

class AuthService {
  async login(request: LoginRequest): Promise<AuthResponse> {
    const response = await authApi.login(request);
    return response.data;
  }

  async refreshToken(
    request: RefreshTokenRequest
  ): Promise<TokenResponse> {
    const response = await authApi.refreshToken(request);
    return response.data;
  }

  async logout(): Promise<void> {
    await authApi.logout();
  }

  async getCurrentUser(): Promise<UserResponse> {
    const response = await authApi.getCurrentUser();
    return response.data;
  }

  async register(
    request: RegisterRequest
  ): Promise<RegisterResponse> {
    const response = await authApi.register(request);
    return response.data;
  }
}

export const authService = new AuthService();
